pandas/core/generic.py

```diff
@@ -1906,23 +1906,54 @@ def to_pickle(self, path, compression='infer',
         Parameters
         ----------
         path : string
-            File path
+            File path where the pickled object will be stored.
         compression : {'infer', 'gzip', 'bz2', 'xz', None}, default 'infer'
-            a string representing the compression to use in the output file
+            A string representing the compression to use in the output file. By
+            default, infers from the specified path.
```

Comment:
"infers from the specified file extension" may be?

Reply:
yes pythonista.